browser/fxr/content/fxrui.js

```diff
@@ -431,3 +438,34 @@ function finishPrompt() {
     currentPermissionRequest = null;
   }
 }
+
+// Called when navigating to another page (aState == null) or when an event
+// fires that the UI should be updated
+function permissionUpdate(aState) {  
+  // For now, only supporting microphone
+  let icon = document.getElementById("ePermsMicrophone");
+
+  // First, remove any classes that make the icon visible
+  icon.classList.remove("urlbar_device_microphone_allowed");
+  icon.classList.remove("urlbar_device_microphone_blocked");
+
+  // Next, determine if the icon should be visible and how
+  if (aState && aState.hasOwnProperty("sharing")
```

Comment:
Perhaps just `aState?.sharing === "microphone"`?

Reply:
oooh, fancy! thank you